packages/puppeteer-core/src/api/Realm.ts

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright 2023 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {TimeoutSettings} from '../common/TimeoutSettings.js';
+import {
+  EvaluateFunc,
+  EvaluateFuncWith,
+  HandleFor,
+  InnerLazyParams,
+  NodeFor,
+} from '../common/types.js';
+import {getPageContent, withSourcePuppeteerURLIfNone} from '../common/util.js';
+import {TaskManager, WaitTask} from '../common/WaitTask.js';
+import {assert} from '../util/assert.js';
+
+import {ClickOptions, ElementHandle} from './ElementHandle.js';
+import {KeyboardTypeOptions} from './Input.js';
+import {JSHandle} from './JSHandle.js';
+
+/**
+ * @internal
+ */
+export abstract class Realm implements Disposable {
```

Comment:
I think we should rename this as Realm is `BiDi's` `ExecutionContext`. Given we are trying to move to more spec(y) names I would go for `Sandbox` as it represents that its a Realm but can be isolated or main

Reply:
I can do this in a separate PR (refactoring in this part of the CL chain will be a headache and it's internal so it should be okay to do).

However, I think `Realm` makes sense here. It's well-defined in JS (see https://tinyurl.com/fvwnyd93 and https://tc39.es/ecma262/#sec-code-realms)

Reply:
Our wrapper around ECMA Realm's are `ExecutionContext` and `(BiDi)Realm`. The Sandbox is an abstraction on top of them that uses Realm and helpers (`this.document`) to simplify work.

Reply:
Sure, but we should really just name everything Realm because that's what javascript defines a sandbox as.